src/pages/Overview/OverviewToolbar.tsx

```diff
@@ -114,24 +145,44 @@ class OverviewToolbar extends React.Component<Props, State> {
           <ToolbarDropdown
             id="overview-duration"
             disabled={false}
-            handleSelect={this.updateDuration}
+            handleSelect={this.props.setDuration}
```

Comment:
I think something is missing here, it's going to dispatch the set-duration action but it doesn't trigger a refresh anymore (when the duration is changed, health data should be reloaded immediately)

Reply:
Does it not get handled here: https://github.com/kiali/kiali-ui/pull/831/files#diff-146a26206017a43c7930bad618f749c6R122 ?

Reply:
Does it not get handled here? https://github.com/kiali/kiali-ui/pull/831/files#diff-146a26206017a43c7930bad618f749c6R122

Reply:
Hmmm, in a quick test it doesn't look that the filter change is triggering the callback, I'll look more closely.

Reply:
No, `StatefulFilters` is only responsible for the filter dropdown and its associated field, but not the other dropdowns. You might have been confused because this `ToolbarDropdown` is declared as a child of  `StatefulFilters`, but they are actually independent, it's just something necessary for how the DOM is organized, but nothing more.

Reply:
OK, I see, that is only invoked when adding an actual filter, not changing a dropdown.

Reply:
I'll make a change.